service/src/main/java/uk/nhs/adaptors/gp2gp/ehr/request/EhrExtractAckHandler.java

```diff
@@ -0,0 +1,79 @@
+package uk.nhs.adaptors.gp2gp.ehr.request;
+
+import java.time.Instant;
+import java.util.List;
+import java.util.stream.Collectors;
+import java.util.stream.IntStream;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+import org.w3c.dom.Document;
+import org.w3c.dom.NodeList;
+
+import lombok.RequiredArgsConstructor;
+import lombok.SneakyThrows;
+import lombok.extern.slf4j.Slf4j;
+import uk.nhs.adaptors.gp2gp.common.service.XPathService;
+import uk.nhs.adaptors.gp2gp.ehr.EhrExtractStatusService;
+import uk.nhs.adaptors.gp2gp.ehr.model.EhrExtractStatus.EhrReceivedAcknowledgement;
+import uk.nhs.adaptors.gp2gp.ehr.model.EhrExtractStatus.EhrReceivedAcknowledgement.EhrReceivedAcknowledgementBuilder;
+import uk.nhs.adaptors.gp2gp.ehr.model.EhrExtractStatus.EhrReceivedAcknowledgement.ErrorDetails;
+import uk.nhs.adaptors.gp2gp.mhs.InvalidInboundMessageException;
+
+@Service
+@Slf4j
+@RequiredArgsConstructor(onConstructor = @__(@Autowired))
+public class EhrExtractAckHandler {
+    private static final String ACK_TYPE_CODE_XPATH = "//MCCI_IN010000UK13/acknowledgement/@typeCode";
+    private static final String ACK_DETAILS_XPATH = "//MCCI_IN010000UK13/acknowledgement/acknowledgementDetail/code";
+    private static final String MESSAGE_REF_XPATH = "//MCCI_IN010000UK13/acknowledgement/messageRef/id/@root";
+    private static final String MESSAGE_ID_ROOT_XPATH = "//MCCI_IN010000UK13/id/@root";
+    private static final String ERROR_CODE_XPATH = "//justifyingDetectedIssueEvent/code";
+    private static final String ACK_OK_CODE = "AA";
+    private static final String ACK_BUSINESS_ERROR_CODE = "AE";
+    private static final String ACK_REJECTED_CODE = "AR";
+    private static final String CODE_ATTRIBUTE = "code";
+    private static final String DISPLAY_ATTRIBUTE = "displayName";
+
+
+    private final XPathService xPathService;
+    private final EhrExtractStatusService ehrExtractStatusService;
+
+    @SneakyThrows
+    public void handle(String conversationId, Document document) {
```

Comment:
Should we create some tests for this method?

Reply:
Added tests for negative scenario. All other are covered in EhrAcknowledgementTest integration test.